src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Debugging/DefaultRazorProximityExpressionResolver.cs

```diff
@@ -24,15 +24,27 @@ internal class DefaultRazorProximityExpressionResolver : RazorProximityExpressio
         private readonly FileUriProvider _fileUriProvider;
         private readonly LSPDocumentManager _documentManager;
         private readonly LSPProjectionProvider _projectionProvider;
-        private readonly VisualStudioWorkspaceAccessor _workspaceAccessor;
+        private readonly CodeAnalysis.Workspace _workspace;
         private readonly MemoryCache<CacheKey, IReadOnlyList<string>> _cache;
 
         [ImportingConstructor]
         public DefaultRazorProximityExpressionResolver(
```

Comment:
Great question!

So the breakpoint resolver controls what happens when you set a breakpoint in the Razor editor OR you type around a breakpoint. Basically it ensures that the red highlight in VS is highlighting the correct portion, aka: 
![image](https://user-images.githubusercontent.com/2008729/110705790-b9b57600-81ab-11eb-93e2-7643fca8cbb2.png)

Now as for the proximity expression resolver, that comes into play when you hit a breakpoint. So when you hit a breakpoint several windows get populated with data, one of which is the "Autos" and "Locals" window:
![image](https://user-images.githubusercontent.com/2008729/110706023-14e76880-81ac-11eb-8946-27f1db48ca2e.png)
The identifiers in these windows are called "expressions" because those expressions can resolve out to be debug-important information. SO in our case the proximity expression resolver takes a location and finds all of the potentially relevant "expressions" to show in those windows.